WeScanTests/VisionRectangleDetectorTests.swift

```diff
@@ -37,27 +67,77 @@ final class VisionRectangleDetectorTests: FBSnapshotTestCase {
     containerLayer.render(in: UIGraphicsGetCurrentContext()!)
 
     let image = UIGraphicsGetImageFromCurrentImageContext()!
+
     UIGraphicsEndImageContext()
 
     let ciImage = CIImage(cgImage: image.cgImage!)
-    let expectation = XCTestExpectation(description: "Detect rectangle")
+    let expectation = XCTestExpectation(description: "Detect rectangle on CIImage")
+
     VisionRectangleDetector.rectangle(forImage: ciImage) { (quad) in
-      
+
       DispatchQueue.main.async {
-        
+
         let resultView = UIView(frame: containerLayer.frame)
         resultView.layer.addSublayer(containerLayer)
-        
+
         let quadView = QuadrilateralView(frame: resultView.bounds)
         quadView.drawQuadrilateral(quad: quad!, animated: false)
         quadView.backgroundColor = UIColor.red
         resultView.addSubview(quadView)
-        
+
         self.FBSnapshotVerifyView(resultView)
         expectation.fulfill()
       }
     }
-    
+
+    wait(for: [expectation], timeout: 3.0)
+  }
+
+  func testCorrectlyDetectsAndReturnsQuadilateralPixelBuffer() {
+
+    let targetSize = CGSize(width: 150, height: 150)
+
+    let containerLayer = CALayer()
+    containerLayer.backgroundColor = UIColor.white.cgColor
+    containerLayer.frame = CGRect(origin: .zero, size: targetSize)
+    containerLayer.masksToBounds = true
+
+    let targetLayer = CALayer()
+    targetLayer.backgroundColor = UIColor.black.cgColor
+    targetLayer.frame = containerLayer.frame.insetBy(dx: 5, dy: 5)
+
+    containerLayer.addSublayer(targetLayer)
+
+    UIGraphicsBeginImageContextWithOptions(targetSize, true, 0.0)
+
+    containerLayer.render(in: UIGraphicsGetCurrentContext()!)
+
+    let image = UIGraphicsGetImageFromCurrentImageContext()!
+
+    UIGraphicsEndImageContext()
+
+    let expectation = XCTestExpectation(description: "Detect rectangle on CVPixelBuffer")
+    if let pixelBuffer = self.buffer(from: image) {
+      VisionRectangleDetector.rectangle(forPixelBuffer: pixelBuffer) { (quad) in
+
+        DispatchQueue.main.async {
+
+          let resultView = UIView(frame: containerLayer.frame)
+          resultView.layer.addSublayer(containerLayer)
+
+          let quadView = QuadrilateralView(frame: resultView.bounds)
+          quadView.drawQuadrilateral(quad: quad!, animated: false)
+          quadView.backgroundColor = UIColor.red
+          resultView.addSubview(quadView)
+
+          self.FBSnapshotVerifyView(resultView)
```

Comment:
```suggestion
          FBSnapshotVerifyView(resultView)
```

Reply:
Just went into this and realized that self is required in this closure, as it's async and a completion block

Reply:
My bad, missed that!